python/lsst/daf/persistence/repositoryCfg.py

```diff
@@ -0,0 +1,152 @@
+        """
+        d = loader.construct_mapping(node)
+        cfg = RepositoryCfg(root=d['_root'], mapper=d['_mapper'], mapperArgs=d['_mapperArgs'], 
+                            parents=d['_parents'], isLegacyRepository=d['_isLegacyRepository'])
+        return cfg
+
+    def __eq__(self, other):
+        if not other:
+            return False
+        return self._root == other._root and \
+               self.mapper == other._mapper and \
+               self.mapperArgs == other._mapperArgs and \
+               self.parents == other._parents and \
+               self._isLegacyRepository == other._isLegacyRepository
+
+    @property
+    def root(self):
+        return self._root
+
+    @root.setter
+    def root(self, root):        
+        if root is not None and self._root is not None:
+            raise RuntimeError("Explicity clear root (set to None) before changing the value of root.")
+        self._root = root
+
+    @property
+    def mapper(self):
+        return self._mapper
+
+    @property
+    def mapperArgs(self):
+        return self._mapperArgs
+    
+    @mapperArgs.setter
+    def mapperArgs(self, newDict):
+        self._mapperArgs = newDict
+
+    @property
+    def parents(self):
+        return self._parents
+
+    def addParents(self, newParents):
+        newParents = listify(newParents)
+        for newParent in newParents:
+            if newParent not in self._parents:
+                self._parents.append(newParent)
+
+    @property
+    def isLegacyRepository(self):
+        return self._isLegacyRepository
+
+    @staticmethod
+    def makeFromArgs(repositoryArgs, parents):
+        cfg = RepositoryCfg(root=repositoryArgs.root, 
+                            mapper = repositoryArgs.mapper, 
+                            mapperArgs = repositoryArgs.mapperArgs,
+                            parents=parents,
+                            isLegacyRepository=repositoryArgs.isLegacyRepository)
+        return cfg
+
+    def matchesArgs(self, repositoryArgs):
+        if repositoryArgs.root is not None and self._root != repositoryArgs.root:
+            return False
+        if repositoryArgs.mapper is not None:
+            if inspect.isclass(self._mapper):
+                if not inspect.isclass(repositoryArgs.mapper):
+                    return False
+                if self._mapper != repositoryArgs.mapper:
+                    return False
+            else:
+                if type(self._mapper) != type(repositoryArgs.mapper):
+                    return False
+        # check mapperArgs for any keys in common and if their value does not match then return false.
+        if self._mapperArgs is not None and repositoryArgs.mapperArgs is not None:
+            for key in set(self._mapperArgs.keys()) & set(repositoryArgs.mapperArgs):
+                if self._mapperArgs[key] != repositoryArgs.mapperArgs[key]:
+                    return False
+        return True
```

Comment:
Another option (both here and in the hashability discussion elsewhere on this PR) is to simply reduce the ways a mapper can be specified.  If the the task framework specifying different mappers in different ways is effectively being considered a bug here, we should just disallow some of those ways up front and force the task framework to be consistent.  Of course, if some of these mapper arg types are just for backwards compatibility, that's different, but I didn't get that impression.